src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -128,7 +135,27 @@ export class OrganizationIntegrations extends AsyncComponent<
 
     const list = this.sortIntegrations(combined);
 
-    this.setState({list, displayedList: list}, () => this.trackPageViewed());
+    const allFeatureDescriptions: IntegrationFeature[] = list.reduce(
+      (acc, integration) => {
+        if (isSentryApp(integration)) {
+          const array = ['internal', 'unpublished'].includes(integration.status)
+            ? [{featureGate: integration.status, description: ''}]
+            : integration.featureData;
+          return acc.concat(...array);
```

Comment:
I've typically written this as `[...acc, ...array]` to avoid mutating the accumulator, and return a new instance, just as a good habit.

Reply:
@EvanPurkhiser `.concat` does not mutate `acc`. It returns a new array with a shallow copy of 'acc'. [Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat)